src/AppContainer.tsx

```diff
@@ -123,6 +126,26 @@ const AppContainer: FunctionComponent<Props> = ({ onSetWorkspaceInfo, onReconnec
         onSetWorkspaceInfo(workspaceInfo)
     })
 
+    const workspaceSubfeedForReducer = useMemo((): AppendOnlyLog => {
+        return {
+            appendMessage: (msg: any) => {
+                workspaceSubfeed.appendMessage({action: msg})
+            },
+            allMessages: () => (workspaceSubfeed.allMessages().map(m => (m.action || {}))),
+            onMessage: (callback: (msg: any) => void) => {
+                workspaceSubfeed.onMessage((m: any) => {
+                    callback(m.action || {})
+                })
+            }
+        }
+    }, [workspaceSubfeed])
+
+    const [workspace, workspaceDispatch] = useFeedReducer<WorkspaceState, WorkspaceAction>(
```

Comment:
Here's the workspace/workspaceDispatch pair... the useFeedReducer allows us to dispatch actions to the feed (on the server) and then they get dispatched to the workspace reducer once they come back to the client as subfeed updates.